pilot/pkg/bootstrap/server.go

```diff
@@ -1249,3 +1262,27 @@ func (s *Server) addFileWatcher(file string, callback func()) {
 		}
 	}()
 }
+
+func (s *Server) waitForCacheSync(stop <-chan struct{}) bool {
+	if !pilot.EnableWaitCacheSync {
```

Comment:
you dont need this flag.. this is essential functionality

Reply:
you mean to move out of this function?

Reply:
I meant the whole feature flag seems unnecessary since this is a reproducible bug that occurs very frequently.. sigh

Reply:
But the code change has zero tests and high risk. So I would say a flag or tests are quite necessary.